torch/csrc/jit/passes/onnx.cpp

```diff
@@ -448,14 +444,6 @@ void NodeToONNX(
         env,
         operator_export_type);
 
-    // Find new nodes that have been created by _run_symbolic_function and
-    // propagate metadata
-    for (auto node : g->nodes()) {
-      if (nodes_before.find(node) == nodes_before.end()) {
-        node->copyMetadata(n);
```

Comment:
This was introduced in #45256 and the main motivation is to propagate "sourceRange" and "scope" information.

The node level copyMetadata does not copy shape and type

https://github.com/pytorch/pytorch/blob/b4c810491ea726df4d3857a1eb8d46e9edc212f5/torch/csrc/jit/ir/ir.h#L436-L443

Indeed this is introducing quadratic cost. If profiling also suggests this being a hotspot, maybe we should add extra state to make the "new nodes" added during `_run_symbolic_function` retrievable.

Reply:
Now I remember it's the sourceRange and scope info and not the shape/type. I got it mixed up. OK, I'll look into adding a state to make all new nodes inherit this. It can be sort of a "set default sourceRange and scope".

Reply:
I have now pushed an update where the new nodes are being tracked by the `GraphContext`. @BowenBao Let me know what you think.